### PR TITLE
Added examples to mark visibility section

### PR DESCRIPTION
**Pull request by nemild**

### CODE REVIEW

File: smart-contracts.md [hunk 1]
```diff
@@ -172,10 +172,28 @@ Fixed point types are currently being implemented in Solidity (ie fixed, ufixed,
...
+```
+// bad
+uint x; // the default is public protected, but it should be made explicit
+function transfer() {
+
+}
+
+// good
+uint public protected y;
+function transfer public () {
```
Comment (simondlr) [inline smart-contracts.md]: Wrong syntax. Please test this. The brackets should be on transfer() public.

File: smart-contracts.md [hunk 2]
```diff
@@ -172,10 +172,28 @@ Fixed point types are currently being implemented in Solidity (ie fixed, ufixed,
...
 
+```
+// bad
+uint x; // the default is public protected, but it should be made explicit
+function transfer() {
+
+}
+
+// good
+uint public protected y;
```
Comment (simondlr) [inline smart-contracts.md]: protected is not currently a keyword. http://solidity.readthedocs.io/en/latest/contracts.html?highlight=private#visibility-and-accessors

**Comment by nemild** [comment]
Thanks, updated. Saw the *Visibility Specifiers* section [here](https://github.com/ethereum/wiki/wiki/Solidity-Features) which talks about the `protected` keyword, but doesn't seem to compile.

**Comment by simondlr** [comment]
Ah yeah. I keep bumping into that page as well. :D

File: smart-contracts.md [hunk 3]
```diff
@@ -172,9 +172,27 @@ Fixed point types are currently being implemented in Solidity (ie fixed, ufixed,
...
 
-### Visibility
+#### Explicitly mark visibility in functions and state variables
 
-Explicitly label the visibility of functions and state variables. Functions can be specified as being external, public, internal or private. For state variables, external is not possible.
+Explicitly label the visibility of functions and state variables. Functions can be specified as being `external`, `public`, `internal` or `private`. For state variables, `external` is not possible.
+
+```
+// not great
+uint x; // the default is public for state variables, but it should be made explicit
```
Comment (simondlr) [inline smart-contracts.md]: This is incorrect. Default for state variables is private.

File: smart-contracts.md [hunk 4]
```diff
@@ -172,9 +172,27 @@ Fixed point types are currently being implemented in Solidity (ie fixed, ufixed,
...
+
+}
+
+// good
+uint public y;
+function transfer() public {
+
+}
+
+function internalAction() private {
```
Comment (simondlr) [inline smart-contracts.md]: I would probably change "private" to "internal". Private is different to internal and might be confusing here.

**Comment by nemild** [comment]
Fixed - let's go through all of these in our chat to make sure that there is no errors. Additionally, we may want to add an explanation for each keyword in the future.

**Pull request merged**